### PR TITLE
Table responsive cards

### LINKED ISSUE #25: Build responsive table cells pattern
Design spec: https://github.com/ubuntudesign/vanilla-dashboard-theme-design/tree/master/Table%20responsive

### PR DESCRIPTION
**Pull request by richmccartney**
## Done

- Created new table card style for mobile
- Styled contextual menus to follow the same style as per MAAS

## QA

- Pull down the branch and do `./run` in the terminal
- Check the responsive tables HTML and SCSS
- Check against designs https://github.com/ubuntudesign/vanilla-dashboard-theme-design/blob/master/Table%20responsive/table-responsive.png

## Details

Fixes #25

### CODE REVIEW

**Comment by kwm14** [comment]
- On `$breakpoint-medium`and `$breakpoint-small` why are you presented with 4 links rather than the 1 link seen in `$breakpoint-large` ?
- On `$breakpoint-medium`and `$breakpoint-small` title 'Name' is spelt in lower case

File: scss/_patterns_table-responsive.scss [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        &__link {
+          border: {
+            color: $color-mid-light;
+            radius: .125rem;
+            style: solid;
+            width: 1px;
+          }
+          box-sizing: border-box;
+          cursor: pointer;
+          color: $color-x-dark;
```
Comment (barrymcgee) [inline scss/_patterns_table-responsive.scss]: This line should be above `cursor: pointer;` for linting.

File: scss/_patterns_table-responsive.scss [hunk 2]
```diff
@@ -0,0 +1,115 @@
+@import 'settings';
+
+@mixin theme-p-table-responsive {
+  .p-table--responsive {
+    @media screen and (max-width: $breakpoint-medium) {
+      thead {
```
Comment (barrymcgee) [inline scss/_patterns_table-responsive.scss]: Except in exceptional circumstances, we've previously stuck to a rule that any rules in the `_patterns_` layer of the stack only targets classes, not elements directly. Are these rules generic enough to be moved up into a base layer - or further upstream to Vanilla?

Comment (richmccartney) [inline scss/_patterns_table-responsive.scss]: So this an issue I was torn about. As @Lukewh has done in a similar vein with table sorting and I can see something similar happening with interactive tables. Looking at a few other frameworks (Bootstrap / Foundation) they've kept any modification all on the `<table>` element. 

Regarding pushing them upstream. I really have a thought that this framework should entirely be pushed upstream and quite possibly put in the layer with the option to have a card view or overflow scroll version of the table where applicable.

**Comment by kwm14** [comment]
Looking good!

**Comment by richmccartney** [comment]
@anthonydillon / @barrymcgee can I get a follow up on this please?

**Comment by anthonydillon** [comment]
@richmccartney @barrymcgee is reviewing this one.

File: examples/patterns/table-responsive/table-responsive.html [hunk 3]
```diff
@@ -0,0 +1,51 @@
+---
+layout: default
+title: Table responsive
```
Comment (barrymcgee) [inline examples/patterns/table-responsive/table-responsive.html]: This is still called 'Table responsive' - please change to just 'Table'

File: examples/patterns/table-responsive/table-responsive.html [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        </tr>
+        <tr role="row">
+            <td role="rowheader" aria-label="Name">Apple</td>
+            <td role="gridcell" aria-label="Users" class="u-align--right">9</td>
+            <td role="gridcell" aria-label="Units" class="u-align--right">17</td>
+            <td role="gridcell"class="u-align--right">
+            </td>
+        </tr>
+    </tbody>
+</table>
```
Comment (barrymcgee) [inline examples/patterns/table-responsive/table-responsive.html]: Indentation in this file is 4 spaces when it should be 2 please.

File: scss/_patterns_table-responsive.scss (hunk 2 above)
Comment (barrymcgee) [inline scss/_patterns_table-responsive.scss]: For consistency, I'd prefer we only styled class names, not elements. Second opinion @anthonydillon ?

Comment (richmccartney) [inline scss/_patterns_table-responsive.scss]: Hmmm we've now landed two branches that goes against this ideal. @anthonydillon mentioned that the responsive table possibly should be set in base, do you feel this is correct approach now? 

Also if we go down this route table sorting / expanding rows and responsive tables will start to get huge class names. I feel some exceptions to the rules should be made for this area of the framework.

Comment (Lukewh) [inline scss/_patterns_table-responsive.scss]: My opinion is that adding a class here is a bit overkill. The context of `thead` is within a table and unless the class is going to target other elements we shouldn't pollute the markup. Tables contain special elements that only exist within a table, is a modification is affecting only a table, the classes should be applied to the table element.
Maybe we should discuss this in dev catchup?

File: examples/patterns/table-responsive/table-responsive.html (hunk 3 above)
Comment (richmccartney) [inline examples/patterns/table-responsive/table-responsive.html]: We've got Table expanding and sortable as titles for others, I don't understand why we would just have `table` as it's not really explaining the content on the page. In other areas of VF.. we've got titles like Form validation, should that be changed to form?

Comment (barrymcgee) [inline examples/patterns/table-responsive/table-responsive.html]: Every pattern within Vanilla is inherently responsive - should be append responsive to every pattern title?

Comment (anthonydillon) [inline examples/patterns/table-responsive/table-responsive.html]: Let's relabel this pattern to represent what the pattern does not its styling.

**Pull request merged**